observatory-dags/observatory/dags/workflows/onix_workflow.py

```diff
@@ -0,0 +1,509 @@
+        """
+        return str()
+
+    @property
+    def extract_folder(self) -> str:
+        """Not used.
+        :return: Empty string.
+        """
+        return str()
+
+    def cleanup(self):
+        """Delete all files and folders associated with this release.
+        :return: None.
+        """
+        shutil.rmtree(self.transform_folder)
+
+
+class OnixWorkflow(Telescope):
+    """This workflow telescope:
+    1. [Not implemented] Creates an ISBN13-> internal identifier lookup table.
+    2. Creates an ISBN13 -> WorkID lookup table.
+      a. Aggregates Product records into Work clusters.
+      b. Writes the lookup table to BigQuery.
+      c. Writes an error table to BigQuery.
+    3. Create an ISBN13 -> Work Family ID lookup table.  Clusters editions together.
+      a. Aggregate Works into Work Families.
+      b. Writes the lookup table to BigQuery.
+    4. Create OAEBU intermediate tables.
+      a. For each data partner, create new tables in oaebu_intermediate dataset where existing tables are augmented
+         with work_id and work_family_id columns.
+    """
+
+    DAG_ID_PREFIX = "onix_workflow"
+
+    def __init__(
+        self,
+        *,
+        org_name: str,
+        gcp_project_id: str,
+        gcp_bucket_name: str,
+        telescope_sensor: ExternalTaskSensor,
+        onix_dataset_id: str = "onix",
+        onix_table_id: str = "onix",
+        dag_id: Optional[str] = None,
+        start_date: Optional[datetime] = datetime(2021, 3, 28),
+        schedule_interval: Optional[str] = "@weekly",
+        catchup: Optional[bool] = False,
+        data_partners: List[OaebuPartners] = list(),
+    ):
+        """Initialises the workflow object.
+        :param org_name: Organisation name.
+        :param gcp_project_id: Project ID in GCP.
+        :param gcp_bucket_name: GCP bucket name to store files.
+        :param telescope_sensor: Telescope sensor to monitor.
+        :param onix_dataset_id: GCP dataset ID of the onix data.
+        :param onix_table_id: GCP table ID of the onix data.
+        :param dag_id: DAG ID.
+        :param start_date: Start date of the DAG.
+        :param schedule_interval: Scheduled interval for running the DAG.
+        :param catchup: Whether to catch up missed DAG runs.
+        :param data_partners: OAEBU data sources.
+        """
+
+        self.dag_id = dag_id
+        if dag_id is None:
+            self.dag_id = make_dag_id(self.DAG_ID_PREFIX, org_name)
+
+        self.org_name = org_name
+        self.gcp_project_id = gcp_project_id
+        self.gcp_bucket_name = gcp_bucket_name
+        self.onix_dataset_id = onix_dataset_id
+        self.onix_table_id = onix_table_id
+
+        # Initialise Telesecope base class
+        super().__init__(
+            dag_id=self.dag_id, start_date=start_date, schedule_interval=schedule_interval, catchup=catchup
+        )
+
+        # Wait on ONIX telescope to finish
+        self.add_sensor(telescope_sensor)
```

Comment:
I was trying to avoid having a dependency on Observatory API classes by passing in a sensor pre-constructed. Would you prefer the org name gets passed to the telescope?